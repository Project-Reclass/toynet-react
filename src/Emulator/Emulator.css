.emulator {
  display: flex;
  max-width: 1920px;
  margin: auto;
}