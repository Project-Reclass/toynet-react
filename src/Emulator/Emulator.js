import React from 'react';
import Instructions from './Instructions/Instructions';
import Tabs from './Tabs/Tabs';
import Visuals from './Visuals/Visuals';

const Emulator = ({ panelData }) => {
  return (
    <div style={{display: 'flex'}}>
      <Instructions panelData={panelData} />
      <div>
        <Tabs />
        <Visuals />
      </div>
    </div>
  );
};

export default Emulator;

