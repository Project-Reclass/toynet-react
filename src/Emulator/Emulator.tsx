/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/
import React from 'react';
import { Grid } from '@chakra-ui/core';

import Visuals from './Visuals';
import Instructions from './Instructions';
import DialogueBox from './DialogueBox';
import Console from './Console';
import styled from '@emotion/styled-base';
import DrawerProvider from '../common/providers/DrawerProvider';

const data = {
  'id': 1,
  'submoduleNumber': 1,
  'submoduleName': 'Modifying Topology',
  'objective': 'Add and remove devices and their connections.',
  'tasks': [
    'Add a Host (h3)',
    'Attach h3 to s2',
    'Detach h3 from s2',
    'Remove h2',
  ],
};

const EmulatorGrid = styled(Grid)`
  margin: auto;
  max-width: 1920px;
  height: 100vh;
  max-height: 100vh;
  overflow: hidden;
  padding: 0.789rem;
  grid-template-columns: auto 1fr;
  @media (max-width: 2050px) {
    margin-left: 5rem;
  }
`;

const Emulator = () => {
  return (
    <DrawerProvider>
      <EmulatorGrid gap={2}>
        <Instructions panelData={data} />
        <Grid
          gridTemplateRows={'1fr 1fr'}
          width='100%'
          overflow='hidden'
          gap={2}
        >
          <Visuals />
          <Grid
            height='100%'
            maxH='100%'
            overflow='hidden'
            gap={2}
            gridTemplateColumns={'2fr 1fr'}
          >
            <Console />
            <DialogueBox />
          </Grid>
        </Grid>
      </EmulatorGrid>
    </DrawerProvider>
  );
};

export default Emulator;
