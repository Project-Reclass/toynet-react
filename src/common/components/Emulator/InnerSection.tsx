/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/
import { Box } from '@chakra-ui/react';
import styled from '@emotion/styled';

export const EmulatorInnerSection = styled(Box)`
  background-color: #212529;
  width: 100%;
  height: 100%;
  position: relative;
  flex: 1 1 auto;
  overflow-y: auto;
  padding: 0.549rem;
  margin-bottom: 0.549rem;
`;

export default EmulatorInnerSection;