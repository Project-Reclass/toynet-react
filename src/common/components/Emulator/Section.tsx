/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/
import { Box } from '@chakra-ui/react';
import styled from '@emotion/styled';

interface Props {
  width?: string;
  height?: string;
  minWidth?: string;
  maxWidth?: string;
  padding?: string
}

const EmulatorSection = styled(Box)`
  color: white;
  position: relative;
  background-color: #454950;
  height: ${(props: Props) => props.height || '100%'};
  width: ${(props: Props) => props.width || '100%'};
  border-radius: 10px;
  padding: ${(props: Props) => props.padding || '2vh'};
  display: flex;
  flex-direction: column;
  min-width: ${(props: Props) => props.minWidth};
  max-width: ${(props: Props) => props.maxWidth};
`;

export default EmulatorSection;