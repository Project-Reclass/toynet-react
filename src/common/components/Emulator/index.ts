export { default as EmulatorTitle } from './Title';
export { default as EmulatorInnerSection } from './InnerSection';
export { default as EmulatorSection } from './Section';