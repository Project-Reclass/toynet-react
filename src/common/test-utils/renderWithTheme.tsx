import React, { FC } from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@chakra-ui/core/dist';
import renderer from 'react-test-renderer';

const Wrapper: FC = ({ children }) => (
  <ThemeProvider>{children}</ThemeProvider>
);

export const renderWithTheme = (ui: any) => {
  return render(ui, { wrapper: Wrapper });
};

export const renderTreeWithTheme = (ui: any) => {
  return renderer.create(<Wrapper>{ui}</Wrapper>);
};
