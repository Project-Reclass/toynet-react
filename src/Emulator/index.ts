import Emulator from './Emulator';
import { withEmulatorAndDialogueProvider } from './EmulatorProvider';

export default withEmulatorAndDialogueProvider(Emulator);