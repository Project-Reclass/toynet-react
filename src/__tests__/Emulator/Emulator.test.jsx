import React from 'react';
import renderer from 'react-test-renderer';
import { renderTreeWithTheme } from 'src/common/test-utils/renderWithTheme';

import Emulator from 'src/Emulator/Emulator';

describe('The emulator', () => {
  it('should render and match snapshots', () => {
    const tree = renderTreeWithTheme(<Emulator />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});