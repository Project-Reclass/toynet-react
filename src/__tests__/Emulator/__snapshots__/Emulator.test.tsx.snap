// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The emulator should render and match snapshots 1`] = `
<div>
  <div
    class="css-15mlykw"
  >
    <div
      class="css-8wnozy"
      width="15vw"
    >
      <div
        class="css-1gskj9r"
      >
        <a
          class="css-r59rv1"
          href="/"
        >
          <svg
            class="css-112d3fm"
          >
            backIcon.svg
          </svg>
          <p
            class="css-fzcsno"
          >
            Back to site
          </p>
        </a>
      </div>
      <div
        class="css-1f1lwn6"
      >
        <h2
          class="css-nua0st"
        >
          Module 
          1
        </h2>
        <p
          class="css-fzcsno"
        >
          Modifying Topology
        </p>
      </div>
      <div
        class="css-1f1lwn6"
      >
        <h2
          class="css-nua0st"
        >
          Objective
        </h2>
        <p
          class="css-fzcsno"
        >
          Add and remove devices and their connections.
        </p>
      </div>
      <div
        class="css-1f1lwn6"
        style="display: flex; flex-direction: column; height: 100%;"
      >
        <h2
          class="css-1sdjv6a"
        >
          Tasks
        </h2>
        <div
          class="css-1y5xr9a"
          style="padding: 0.549rem;"
        >
          <div
            class="css-16cm47t"
          >
            <li
              class="css-p5me83"
            >
              Add a Host (h3)
            </li>
            <li
              class="css-p5me83"
            >
              Attach h3 to s2
            </li>
            <li
              class="css-p5me83"
            >
              Detach h3 from s2
            </li>
            <li
              class="css-p5me83"
            >
              Remove h2
            </li>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-jj5y1g"
    >
      <div
        class="css-zy3qbz"
      >
        <div
          class="css-ljj55y"
        >
          <div
            class="react-flow"
          >
            <div
              class="react-flow__renderer react-flow__zoompane"
            >
              <div
                class="react-flow__nodes"
                style="transform: translate(0px,0px) scale(1);"
              />
              <svg
                class="react-flow__edges"
                height="500"
                width="500"
              >
                <defs>
                  <marker
                    class="react-flow__arrowhead"
                    id="react-flow__arrowclosed"
                    markerHeight="12.5"
                    markerWidth="12.5"
                    orient="auto"
                    refX="0"
                    refY="0"
                    viewBox="-10 -10 20 20"
                  >
                    <polyline
                      fill="#b1b1b7"
                      points="-5,-4 0,0 -5,4 -5,-4"
                      stroke="#b1b1b7"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                  </marker>
                  <marker
                    class="react-flow__arrowhead"
                    id="react-flow__arrow"
                    markerHeight="12.5"
                    markerWidth="12.5"
                    orient="auto"
                    refX="0"
                    refY="0"
                    viewBox="-10 -10 20 20"
                  >
                    <polyline
                      fill="none"
                      points="-5,-4 0,0 -5,4"
                      stroke="#b1b1b7"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1.5"
                    />
                  </marker>
                </defs>
                <g
                  transform="translate(0,0) scale(1)"
                />
              </svg>
              <div
                class="react-flow__pane"
              />
            </div>
            <div
              class="css-1qjvdnt"
            >
              <button
                class="css-1ypc82u"
                data-testid="emulator-add-host"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="css-yxiis9"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                    fill="currentColor"
                  />
                </svg>
                Host
              </button>
              <button
                class="css-w42zvo"
                data-testid="emulator-add-switch"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="css-yxiis9"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                    fill="currentColor"
                  />
                </svg>
                Switch
              </button>
              <button
                class="css-1470nqk"
                data-testid="emulator-add-router"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="css-yxiis9"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                    fill="currentColor"
                  />
                </svg>
                Router
              </button>
            </div>
            <div
              class="react-flow__controls css-1if3jds"
            >
              <div
                class="react-flow__controls-button react-flow__controls-zoomin"
              >
                <svg
                  viewBox="0 0 32 32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M32 18.133H18.133V32h-4.266V18.133H0v-4.266h13.867V0h4.266v13.867H32z"
                  />
                </svg>
              </div>
              <div
                class="react-flow__controls-button react-flow__controls-zoomout"
              >
                <svg
                  viewBox="0 0 32 5"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h32v4.2H0z"
                  />
                </svg>
              </div>
              <div
                class="react-flow__controls-button react-flow__controls-fitview"
              >
                <svg
                  viewBox="0 0 32 30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.692 4.63c0-.53.4-.938.939-.938h5.215V0H4.708C2.13 0 0 2.054 0 4.63v5.216h3.692V4.631zM27.354 0h-5.2v3.692h5.17c.53 0 .984.4.984.939v5.215H32V4.631A4.624 4.624 0 0027.354 0zm.954 24.83c0 .532-.4.94-.939.94h-5.215v3.768h5.215c2.577 0 4.631-2.13 4.631-4.707v-5.139h-3.692v5.139zm-23.677.94a.919.919 0 01-.939-.94v-5.138H0v5.139c0 2.577 2.13 4.707 4.708 4.707h5.138V25.77H4.631z"
                  />
                </svg>
              </div>
              <div
                class="react-flow__controls-button react-flow__controls-interactive"
              >
                <svg
                  viewBox="0 0 25 32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21.333 10.667H19.81V7.619C19.81 3.429 16.38 0 12.19 0c-4.114 1.828-1.37 2.133.305 2.438 1.676.305 4.42 2.59 4.42 5.181v3.048H3.047A3.056 3.056 0 000 13.714v15.238A3.056 3.056 0 003.048 32h18.285a3.056 3.056 0 003.048-3.048V13.714a3.056 3.056 0 00-3.048-3.047zM12.19 24.533a3.056 3.056 0 01-3.047-3.047 3.056 3.056 0 013.047-3.048 3.056 3.056 0 013.048 3.048 3.056 3.056 0 01-3.048 3.047z"
                  />
                </svg>
              </div>
            </div>
            <svg
              class="react-flow__background"
              style="width: 100%; height: 100%;"
            >
              <pattern
                height="16"
                id="pattern"
                patternUnits="userSpaceOnUse"
                width="16"
                x="0"
                y="0"
              >
                <circle
                  cx="0.25"
                  cy="0.25"
                  fill="#aaa"
                  r="0.5"
                />
              </pattern>
              <rect
                fill="url(#pattern)"
                height="100%"
                width="100%"
                x="0"
                y="0"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="css-jhr2ok"
      >
        <div
          class="css-1fmka3o"
        >
          <h2
            class="css-ix0oza"
          >
            Console
          </h2>
          <div
            class="css-1y5xr9a"
          />
        </div>
        <div
          style="z-index: 0; overflow: hidden;"
        >
          <div
            class="css-1fmka3o"
          >
            <h2
              class="css-ix0oza"
            >
              Actions
            </h2>
            <div
              class="css-1y5xr9a"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
